bot/cogs/sudo.py

```diff
@@ -45,27 +42,34 @@ def get_uptime(self) -> str:
         minutes, seconds = divmod(rem, 60)
         days, hours = divmod(hours, 24)
         if days:
-            str = f"{days} days, {hours} hr, {minutes} mins, and {seconds} secs"
+            formatted = f"{days} days, {hours} hr, {minutes} mins, and {seconds} secs"
         else:
-            str = f"{hours} hr, {minutes} mins, and {seconds} secs"
+            formatted = f"{hours} hr, {minutes} mins, and {seconds} secs"
+        return formatted
 
-        return str
+    @staticmethod
+    async def cog_check(ctx: Context) -> t.Union[bool, None]:
+        """Only devs can use this."""
+        if ctx.author.id in config.devs:
+            return True
+        embed = Embed(description="This is an owner-only command, you cannot invoke this.", color=Color.red())
+        await ctx.send(embed=embed)
```

Comment:
It is a common practice to put `cog_check` on the bottom of the cog.

Reply:
Really ?  Didn't know

Reply:
Oh ! and one thing that needs to be done : sort the commands alphabetically
That'd be much better

Reply:
While in some cases it would certainly provide an easier way to find commands, in others it might cause a lot of problems since there are commands which should logically be together even if not alphabetically.

For example commands like `embed author`, `embed title` and `embed description` are main embed commands which should be kept together while `embed field description`, `embed field title`, ... should also be grouped together, sorting these alphabetically would separate the `title` command from those first 3 and put `field` commands before that which doesn't make sense. We should always prefer to group things logically, not alphabetically.

To do this purely for easier command finding wouldn't be a good idea since it's a lot of work for something which can be done with a simple find function which can be (in most editors) accessed simply by <kbd>Ctrl</kbd>+<kbd>F</kbd>

Reply:
I mean, sort them alphabetically, but keep groups subcommands under the group.
Also, it's not only for the coding : it's also reflected in the help. Currently, the help is very chaotic. Having sorted commands would feel nicer from the user's point of view.

Reply:
Alright I agree, but make sure to make a change to `CONTRIBUTING.md` where you mention this new requirement of sorting commands alphabetically and note there the obvious logical grouping exception

Reply:
I'll do that tomorrow.

Reply:
Ok done, but I thought about the `cog_check` thing : I feel like it's better to put it at the top, because, that way, when you start reading the cog's code, you can right away see that there's a check for all commands, instead of worrying about (for example) an unprotected eval command that is, in fact, protected by a `cog_check` you haven't seen yet.

Reply:
I'm not sure about that, on one hand, yes, it might make it more visible in some cases, but when we take a look at the whole file, there may be some additional classes apart from just the cog and cog is usually the last one (since it depends on the ones above, hence it makes sense to make it the last class), so it's much simpler to go to the end of file (with last class being the cog class) and easily finding `cog_check` right there at the end of the file (with the setup), while searching for the beginning of the Cog class itself will take some time and the `cog_check` will need to be under `__init__` function anyway which means losing additional fast accessibility, I think it being on the bottom makes more sense and would be faster to find, but it's true that you will have to know where to look for it.